onnxruntime/hosting/converter.cc

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <onnx/onnx_pb.h>
+#include "core/common/logging/logging.h"
+#include "core/framework/data_types.h"
+#include "core/framework/environment.h"
+#include "core/framework/framework_common.h"
+#include "core/framework/mem_buffer.h"
+#include "core/framework/ml_value.h"
+#include "core/framework/tensor.h"
+#include "core/framework/tensorprotoutils.h"
+
+#include "onnx-ml.pb.h"
+#include "predict.pb.h"
+
+#include "converter.h"
+
+namespace onnxruntime {
+namespace hosting {
+
+namespace protobufutil = google::protobuf::util;
+
+onnx::TensorProto_DataType MLDataTypeToTensorProtoDataType(const onnxruntime::DataTypeImpl* cpp_type,
```

Comment:
Remove the logger, return the type within each if statement, and make the whole method `constexpr`.

Reply:
Logger removed. Could not make it `constexpr` because of the `GetType()` in the implementation.